tests/foreman/api/test_contentviewversion.py

```diff
@@ -345,6 +346,77 @@ def test_positive_delete_composite_version(self):
         # Make sure that content view version is really removed
         self.assertEqual(len(composite_cv.read().version), 0)
 
+    @upgrade
+    @tier2
+    def test_positive_delete_with_puppet_content(self):
+        """Delete content view version with puppet module content
+
+        :id: cae1164c-6608-4e19-923c-936e75ed807b
+
+        :steps:
+            1. Create a lifecycle environment
+            2. Create a content view
+            3. Add a puppet module to content view
+            4. Publish the content view
+            5. Promote the content view to lifecycle environment
+            6. Remove the content view versions from all lifecycle environments
+            7. Delete the content view version
+
+        :expectedresults: Content view version deleted successfully
+
+        :BZ: 1417081
+
+        :CaseLevel: Integration
+        """
+        org = entities.Organization().create()
+        lce_library = entities.LifecycleEnvironment(
+            organization=org, name=ENVIRONMENT).search()[0].read()
+        lce = entities.LifecycleEnvironment(
+            organization=org, prior=lce_library).create()
+        product = entities.Product(organization=org).create()
+        puppet_repo = entities.Repository(
+            url=FAKE_0_PUPPET_REPO,
+            content_type=REPO_TYPE['puppet'],
+            product=product,
+        ).create()
+        puppet_repo.sync()
+        # create a content view and add the yum repo to it
+        content_view = entities.ContentView(organization=org).create()
+        # get a random puppet module
+        puppet_module = random.choice(
+            content_view.available_puppet_modules()['results']
+        )
+        # add the puppet module to content view
+        entities.ContentViewPuppetModule(
+            author=puppet_module['author'],
+            name=puppet_module['name'],
+            content_view=content_view,
+        ).create()
+        # publish the content view
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 1)
+        content_view_version = content_view.version[0].read()
+        self.assertEqual(len(content_view_version.environment), 1)
+        lce_library = entities.LifecycleEnvironment(
+            id=content_view_version.environment[0].id).read()
+        self.assertEqual(lce_library.name, ENVIRONMENT)
+        # promote content view version to the created lifecycle environment
+        promote(content_view_version, lce.id)
+        content_view_version = content_view_version.read()
+        self.assertEqual(
+            {lce_library.id, lce.id},
+            {lce.id for lce in content_view_version.environment}
+        )
+        # remove the content view versions from all lifecycle environments
+        for env in (lce_library, lce):
```

Comment:
Does it perform `Katello::ContentView.find_by(:name => '<content view name>').versions.first.content_view_puppet_environments.destroy_all` ?

Reply:
The BZ complain of the inability to delete the content view version by the standard sat means eg (API CLI or UI), for me other things are debugging information.

@JacobCallahan why you have verified the BZ that way?

Reply:
@ldjebran Basically, we had to get the Satellite into a bad state that is very hard to occur naturally. In order to do that, we had to manually destroy the puppet environments through the foreman-rake console. This specific bug fix was created to handle the unlikely event that a content view version has a missing puppet environment. 

In my opinion, we don't need to explicitly test this bug, as it would require we delete the puppet environment from a content view through means that we don't readily have available.  However, this test is still valuable to keep, if we don't already cover deleting content views with puppet environments intact.

My recommendation is to keep the test, but not associate with the BZ. I will also drop the test coverage flag, if you agree.